dist-repo-builder/src/main/java/io/ballerina/dist/DistRepoBuilder.java

```diff
@@ -41,70 +49,107 @@
  */
 public class DistRepoBuilder {
 
-    final static String balaGlob = "glob:**/bala.json";
-    final static String jarGlob = "glob:**/*.jar";
-    final static String docGlob = "glob:**/api-docs.json";
+    private final static String balaGlob = "glob:**/*.bala";
+    private final static String jarGlob = "glob:**/*.jar";
+    private static final String BALLERINA_HOME_KEY = "ballerina.home";
 
     public static void main(String args[]) throws IOException {
         System.out.println("Building Distribution Repo ...");
         if (args.length != 1) {
             System.out.println("Invalid Inputs");
             System.exit(1);
         }
-        Path jBalToolsPath = Paths.get(args[0]);
-        Path repo = jBalToolsPath.resolve("repo");
-        System.setProperty("ballerina.home", jBalToolsPath.toString());
+        Path repo = Paths.get(args[0]);
 
         // Find all bala files
         List<Path> balas = findBalas(repo.resolve("bala"));
         // Extract platform libs
         boolean valid = true;
-        // The following list will contain existing docs from ballerina-lang repo
-        List<Path> existingDocs = getExistingDocs(jBalToolsPath.resolve("docs"));
         for (Path bala : balas) {
-            generateDocsFromBala(bala, jBalToolsPath, existingDocs);
-            // following function was put in to validate if bir and jar exists for packed balas
-            valid = valid & validateCache(bala, repo);
-        }
-        if (!valid) {
-            System.exit(1);
+            extractPlatformLibs(bala);
         }
+        generateCaches(balas, repo);
+        //TODO : generate twice to avoid errors
+        generateCaches(balas, repo);
```

Comment:
Once the suggestion in https://github.com/ballerina-platform/ballerina-distribution/pull/1368/files#r663541782 is resolved, remove the duplicated step and check if it is working